go/vt/sqlparser/sql.y

```diff
@@ -98,7 +98,7 @@ func forceEOF(yylex interface{}) {
 %left <empty> '*' '/' '%' MOD
 %left <empty> '^'
 %right <empty> '~' UNARY
-%left <empty> COLLATE
+%right <empty> COLLATE
```

Comment:
I think this should be `left` associative. Or if we want to be more picky, we could use `nonassoc`.